sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/SessionHandler.java

```diff
@@ -26,13 +27,24 @@ public class SessionHandler extends Handler {
     private final String sessionName;
     private final Duration openTimeout;
     private final ReactorDispatcher reactorDispatcher;
+    private final AmqpMetricsProvider metricsProvider;
 
+    /**
+     * @deprecated use {@link SessionHandler#SessionHandler(String, String, String, ReactorDispatcher, Duration, AmqpMetricsProvider)} instead.
+     */
+    @Deprecated
```

Comment:
Because this is non-public API, and if you want people to always use the other API, you could consider just removing this constructor.

Reply:
Same here, this API is used in EH/SB and I have to keep it until at least EH/SB are updated to use it and even then for some time to not break on version mismatch